plugin/utils/shell.py

```diff
@@ -114,51 +114,51 @@ def generate_rig_edit():
     armature = bpy.context.object
     
     #Iterate over every bone in the armature.
-    for i in armature.pose.bones:
+    for p_bone in armature.pose.bones:
         #Check if the bones have been scaled
         #Update locations
         bpy.context.view_layer.update()
         
         #Check that no bones are scaled
-        if i.scale != Vector((1,1,1)):
+        if p_bone.scale != Vector((1,1,1)):
             #Reset scale
-            #i.scale = Vector((1,1,1))
-            print("WARNING: "+str(i.name)+" HAD SCALE, CANCELLING")
-            msgs.append(f"Warning: {str(i.name)} had scale transforms. Reset scale for all bones and try again.")
+            #p_bone.scale = Vector((1,1,1))
+            logging.info("WARNING: "+str(p_bone.name)+" HAD SCALE, CANCELLING")
+            msgs.append(f"Warning: {str(p_bone.name)} had scale transforms. Reset scale for all bones and try again.")
             return msgs
         
         #Find all bones with transforms.
-        if (i.location != Vector((0, 0, 0)) or i.rotation_quaternion != Quaternion((1,0,0,0)) or i.scale != Vector((1,1,1))) and i.name[:5] != "NODE_":
+        if (p_bone.location != Vector((0, 0, 0)) or p_bone.rotation_quaternion != Quaternion((1,0,0,0)) or p_bone.scale != Vector((1,1,1))) and not p_bone.name.startswith("NODE_"):
```

Comment:
I think I would separate the finding, creating and the applying into 3 separate loops?

Reply:
if we do that, we can just use apply pose logic and just place our new nodes in the right spots